pkg/util/steps/refreshing.go

```diff
@@ -0,0 +1,83 @@
+package steps
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+// AuthorizationRefreshingAction returns a wrapper Step which will refresh
+// `authorizer` if the step returns an Azure AuthenticationError and rerun it.
+// The step will be retried until `retryTimeout` is hit. Any other error will be
+// returned directly.
+func AuthorizationRefreshingAction(authorizer refreshable.Authorizer, step Step) authorizationRefreshingActionStep {
+	return authorizationRefreshingActionStep{
+		step:       step,
+		authorizer: authorizer,
+	}
+}
+
+type authorizationRefreshingActionStep struct {
+	step         Step
+	authorizer   refreshable.Authorizer
+	retryTimeout time.Duration
+	pollInterval time.Duration
+}
+
+func (s authorizationRefreshingActionStep) run(ctx context.Context, log *logrus.Entry) error {
+	var pollInterval time.Duration
+	var retryTimeout time.Duration
+
+	// If no pollInterval has been set, use a default
+	if s.retryTimeout == time.Duration(0) {
+		retryTimeout = 10 * time.Minute
+	} else {
+		retryTimeout = s.retryTimeout
+	}
+
+	// If no pollInterval has been set, use a default
+	if s.pollInterval == time.Duration(0) {
+		pollInterval = 10 * time.Second
+	} else {
+		pollInterval = s.pollInterval
+	}
```

Comment:
are you keeping the time.Duration fields to ease unit testing?  If so, why not just populate them in the constructor? (concept applies throughout PR)